addons/findAll/findAll.lua

```diff
@@ -282,7 +283,7 @@ function search(query, export)
     end
     
     for id, names in pairs(key_item_names) do
-        if terms_pattern == '' or key_item_names[id].name:find(terms_pattern) then
+        if terms_pattern == '' or key_item_names[id].name:find(terms_pattern) and settings.Tracker.Key_Item_Display == true then
```

Comment:
While this works, the loop as it is is inefficient. I'd rewrite the entire loop to this:

```lua
if settings.KeyitemDisplay then
    for id in pairs(key_item_names) do
        if terms_pattern == '' or key_item_names[id].name:match(terms_pattern) then
            results_key_items:add(id)
        end
    end
end
```